tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/AspNetMvcIntegration.cs

```diff
@@ -111,6 +114,7 @@ internal static Scope CreateScope(ControllerContextStruct controllerContext)
                         areaName = (routeValues?.GetValueOrDefault("area") as string)?.ToLowerInvariant();
                         controllerName = (routeValues?.GetValueOrDefault("controller") as string)?.ToLowerInvariant();
                         actionName = (routeValues?.GetValueOrDefault("action") as string)?.ToLowerInvariant();
+                        resolvedRoute = $"{areaName}/{controllerName}/{actionName}";
```

Comment:
the thing is routeUrl gives {controller}/{action}/{id} but nothing resolved 🤔

Reply:
Sure, but that _is_ the "route". Which AFAICT, is what we're supposed to provide? We're not supposed to replace route parameters as far as I understand it.

Reply:
As I understand it from [here](https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#http.route) it's gotta be the resolved area/controller/action but not the params so in this case would be for example mycontroller/myaction/{id} , id remaining unresolved... but maybe I m getting it wrong...

Reply:
Again, the trouble here in my opinion is that the standardised tag here is not very useful, as it can mean wildly different things to different languages/frameworks. Regarding this occurence, _technically_ area / controller / action _are also parameters_. So potentially we _shouldn't_ replace them, but there's definitely an argument to be made that we should (for the same reasons that we already do that for the resource etc). Question is, what are customers going to use this tag for? 

Nevertheless, if we decide we _do_ want to replace them, we _can't_ do it like this:
```csharp
resolvedRoute = $"{areaName}/{controllerName}/{actionName}";
```

Because that route might not be correct. e.g. My default route might be `/some-prefix/{action=Index}/{id?}` with a specific `controller` specified in the route parameters. The only way to get the "correct, replaced" route is to start with the route, strip out the constraints and defaults (`=Index`, `?` etc), and do replacements for the values we have. Unfortunately, that's very expensive, and IMO, not worth the effort, so I would just duplicate the value we use for `aspnet.route` and `aspnet_core.route`

Reply:
I also think we should duplicate the `aspnet.route` and `aspnet_core.route`. The current specification we have is vague, but so is the [OTEL HTTP server specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions). I don't have too much experience generating traces with the .NET OpenTelemetry SDK, but it seems like they [also set the route](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1661) to `api/{controller}/{action}/{id}` (not resolved), so I'm inclined to replicate that behavior if it seems reasoanble.

Reply:
sounds good will do this